File: ts/02-enmarcando-nombres.ts
function createFrameTs(names: string[]): string {
  const maxLength = Math.max(...names.map(name => name.length));

  const namesWithWithSpaces = names.map(name => {
    const spaces = ' '.repeat(maxLength - name.length);
    return `* ${name}${spaces} *`;
  });

  const border = '*'.repeat(maxLength + 4);

  return [border, ...namesWithWithSpaces, border].join("\n");
}

File: python/02-enmarcando-nombres.py
def createFramePy(names: list[str]) -> string:
  if not names: return ""
  max_length = max(len(name) for name in names)

  framed_names = [f"* {name} {' ' * (max_length - len(name))} *" for name in names]
  border = "*" * (max_length + 4)

  return "\n".join([border, *framed_names, border])
